src/ir/datalog/input_relation_fact.h

```diff
@@ -47,6 +47,15 @@ class InputRelationFact {
                                          }));
   }
 
+  // Returns string representation of a fact that can be used in an input
+  // relations file.
+  std::string ToDatalogFactsFileString() const {
+    return absl::StrJoin(relation_arguments_, ", ",
+                         [](std::string *str, const auto &arg) {
+                           absl::StrAppend(str, arg.ToDatalogString());
+                         });
+  }
```

Comment:
Talked to you previously about this: I don't think `,` is a good choice of separator. For ADTs that have multiple arguments, the parser will get confused. I've been using `;`, which I'd suggest using here as well.

Reply:
Done.